src/main/webapp/scripts/map.js

```diff
@@ -12,82 +12,145 @@
-};
+window.onload = function() {
+  /* Custom Markers */
+  yellowIcon = {
+    url: "/images/yellow-marker.png",
+    scaledSize: new google.maps.Size(30, 40)
+  };
+  blueIcon = {
+    url: "/images/blue-marker.png",
+    scaledSize: new google.maps.Size(30, 40)
+  };
+  purpleIcon = {
+    url: "/images/purple-marker.png", 
+    scaledSize: new google.maps.Size(30, 40)
+  };
+  redIcon = {
+    url: "/images/red-marker.png", 
+    scaledSize: new google.maps.Size(30, 40)
+  };
+  greyIcon = {
+    url: "/images/grey-marker.png",
+    scaledSize: new google.maps.Size(30, 40)
+  }
+  initMap();
+}
 
 /* Set Up Map */
-function initMap() {
-  /* 'Hard Coded' origin, Google UK Pancras Square*/
+function initMap() { 
+  /* 'Hard Coded' origin, Google UK Pancras Square */
   const origin = {
     lat: 51.533364, 
     lng: -0.125777
-  }
+  };
   const map = new google.maps.Map(document.getElementById("map"), {
     center: origin,
     zoom: 15
   });
   map.addListener("click", e => {
     placeMarkerAndPanTo(e.latLng, map);
   });
-  /*Add Hard-Coded Markers */
+  
+  /* Add Hard-Coded Markers */
   addMarker(map);
 }
 
-/* Temporary Function to Hold Marker for Sky Garden */
 function addMarker(map) {
-  const skyGarden = {lat: 51.510881, lng: -0.083751}
-  const marker = new google.maps.Marker({
-    position: skyGarden,
-    map: map,
-    title: "Sky Garden"
-  })
-  marker.addListener("click", () => {
-    document.getElementById("rec-container").style.display = "block";
-    map.setZoom(16);
-    map.setCenter(marker.getPosition());
-  });
+  /* Iterate through stored recommendations to get details */
+  for (i = 0; i < recommendations.length; i++) {
+    const recommendation = recommendations[i];
+    const placeCategory = recommendation[0];
+    const placeName = recommendation[1];
+    const placeLatLng = new google.maps.LatLng(recommendation[2], recommendation[3]);
+    const placeRecommendation = recommendation[4];
+    const marker = new google.maps.Marker({
+      position: placeLatLng,
+      map: map,
+      title: placeName,
+      icon: getColourMarker(placeCategory)
+    });
+    /* Add Listener for Click on Marker */
+    google.maps.event.addListener(marker, "click", () => {
+      /* Update the HTML */
+      document.getElementById("category-header").innerHTML = placeCategory;
+      document.getElementById("category-header").style.backgroundColor = getBackgroundColour(placeCategory);
+      document.getElementById("place-title").innerHTML = placeName;
+      document.getElementById("rec-address").innerHTML = placeLatLng;
+      document.getElementById("place-recommendation").innerHTML = placeRecommendation;
+      document.getElementById("rec-container").style.display = "block";
+      /* Adjust the map settings */
+      map.setZoom(16);
+      map.setCenter(marker.getPosition());
+    });
+  }
+}
+
+function getBackgroundColour(category) {
+  switch(category) {
+    case 'Restaurants':
+      return "#ffba04";
```

Comment:
nit: it would be nice to add an inline comment for each hex value to clarify what this color is :)